### PR TITLE
Handle dictionary as singleton for always_iterable

### LINKED ISSUE #26: ValueError when attempting to access Pod.subpods
#25 is an issue with v4.1.0 rather than with the API.

https://github.com/jaraco/wolframalpha/commit/77bac9ad5bb730b9146a4cc09d1595c54fb6c678 changed https://github.com/jaraco/wolframalpha/blob/34f89546b2ae11766aebedad8560388e55421cf8/wolframalpha/__init__.py#L80 to use [`more_itertools.always_iterable`](https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.always_iterable) rather than [`jaraco.itertools.always_iterable`](https://jaracoitertools.readthedocs.io/en/latest/#jaraco.itertools.always_iterable).

By default, `jaraco.itertools.always_iterable` treats a dictionary as a singleton, whereas `more_itertools.always_iterable` iterates over the dictionary's keys. This causes accessing [`Pod.subpods`](https://wolframalpha.readthedocs.io/en/latest/?badge=latest#wolframalpha.Pod.subpods) to error with `ValueError: dictionary update sequence element #0 has length 1; 2 is required` now, as when a subpod dictionary is passed to `more_itertools.always_iterable`, it will iterate over each of the keys, and when [`Subpod`](https://wolframalpha.readthedocs.io/en/latest/?badge=latest#wolframalpha.Subpod) is initialized, it will attempt to cast each of those keys, rather than the entire subpod dictionary, to a `dict`.

### PR DESCRIPTION
**Pull request by Harmon758**
Fixes #26

### CODE REVIEW

File: wolframalpha/__init__.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def from_doc(cls, doc):
         Load instances from the xmltodict result. Always return
         an iterable, even if the result is a singleton.
         """
-        return map(cls, always_iterable(doc))
+        return map(cls, always_iterable(doc, base_type=dict))
```
Comment (jaraco) [inline wolframalpha/__init__.py:80]: This change removes the base_types of `str` and `bytes`. I wonder if that's okay in this case.

Comment (jaraco) [reply wolframalpha/__init__.py:80]: Regardless, I've merged this commit onto a test that replicates the failure. If it turns out that the other singletons are important, we can address that, but I suspect they're not as this is passed to a `dict` constructor.

**Comment by jaraco** [comment]
Thanks for this. Releasing as 4.1.1.

**Comment by Harmon758** [comment]
> This change removes the base_types of `str` and `bytes`. I wonder if that's okay in this case.
> Regardless, I've merged this commit onto a test that replicates the failure. If it turns out that the other singletons are important, we can address that, but I suspect they're not as this is passed to a `dict` constructor.

I saw that and that was the same conclusion I came to.

Thanks for the quick merge and release!

**Pull request merged**